src/transports/ws/sws.c

```diff
@@ -678,7 +686,76 @@ static void nn_sws_validate_utf8_chunk (struct nn_sws *self)
     return;
 }
 
-static int nn_sws_fail_conn (struct nn_sws *self, int code, char *reason)
+static void nn_sws_acknowledge_close_handshake (struct nn_sws *self)
+{
+    uint8_t *pos;
+    uint16_t close_code;
+    int code_point_len;
+    size_t len;
+
+    len = self->inmsg_current_chunk_len;
+    pos = self->inmsg_current_chunk_buf;
+
+    /*  Peer did not provide a Close Code, so choose our own here. */
+    if (len == 0) {
+        nn_sws_fail_conn (self, NN_SWS_CLOSE_NORMAL, "");
+        return;
+    }
+
+    /*  If the payload is not even long enough for the required 2-octet
+        Close Code, the connection should have been failed upstream. */
+    nn_assert (len >= NN_SWS_CLOSE_CODE_LEN);
+    len -= NN_SWS_CLOSE_CODE_LEN;
+    pos += NN_SWS_CLOSE_CODE_LEN;
+
+    /*  As per RFC 6455 7.1.6, the Close Reason following the Close Code
```

Comment:
While this is required for generation, do we care about actually checking this?  What happens if we just ignore the close code whether well formed or not?  (Be generous in what you accept.... )